src/core/probe/user/bpf/usdt.bpf.c

```diff
@@ -6,20 +6,51 @@
 
 /* Hook placeholder */
 __attribute__ ((noinline))
-int hook0(struct pt_regs *ctx, struct trace_raw_event *event) {
+int hook0(struct user_ctx *ctx, struct trace_raw_event *event) {
 	volatile int ret = 0;
 	if (!ctx || !event)
 		return 0;
 	return ret;
 }
 
+static __always_inline int get_args(struct user_ctx *uctx,
+				     struct pt_regs *ctx)
+{
+	int cnt = bpf_usdt_arg_cnt(ctx);
+	long tmp = 0;
+
+#define get_arg(x)								\
+	if (x < cnt) {								\
```

Comment:
I had a lot of trouble with this part because the verifier would not detect all of the bytes of the structure were initialized and failed later on with an absolutely unrelated error code.
I'll try again now that the code is cleaner.